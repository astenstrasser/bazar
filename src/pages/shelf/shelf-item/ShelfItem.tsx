import React from "react";
import "./ShelfItem.css";
import { Product } from "../../../shared-components/types";
import { useHistory } from "react-router-dom";

function ShelfItem(product: Product): React.ReactElement {
  const history = useHistory();

  const redirectToDetails = (productId: string) => {
    history.push(`/${productId}`);
  };

  return (
    <div
      className="shelf-item"
      data-testid="shelf-item"
      onClick={() => {
        redirectToDetails(product._id);
      }}
    >
      <div className="shelf-item--header">
        <div className="shelf-item--info">
          <h1>{product.name.toUpperCase()} </h1>
          <p>{product.details}</p>
        </div>
        <div className="shelf-item--price">R$ {product.price}</div>
      </div>
      <img
        className="shelf-item--picture"
        src={product.imageUrl}
        alt={product.imageAltText}
      />
    </div>
  );
}

export default ShelfItem;
