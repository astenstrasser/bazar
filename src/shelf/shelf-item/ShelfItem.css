.shelf-item {
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  align-items: center;
  padding: 1rem;
  max-width: 30rem;
  max-height: 30rem;
  background-color: #ffb703;
  box-shadow: 0 0 9px 1px #e7e7e7;
}

@media (max-width: 600px) {
  .shelf-item {
    margin: 1rem 0;
    width: 100%;
  }
}

@media (min-width: 600px) {
  .shelf-item {
    width: 25%;
    margin: 2rem;
  }
}

.shelf-item--header {
  color: #ffffff;
  display: flex;
  align-items: flex-start;
  justify-content: space-between;
  width: 100%;
}

.shelf-item--header h1 {
  margin-top: 0;
}

.shelf-item--info {
  max-width: 70%;
}

.shelf-item--price {
  color: #023047;
  border: 1px;
  width: 5rem;
  height: 5rem;
  border-radius: 4rem;
  background-color: #f4d06f;
  display: flex;
  justify-content: center;
  align-items: center;
  font-weight: 600;
  text-align: center;
}

.shelf-item--picture {
  width: 10rem;
}
