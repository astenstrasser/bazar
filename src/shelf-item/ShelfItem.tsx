import React from "react";
import "./ShelfItem.css";

interface Item {
  name: string;
  price: number;
  description: string;
  image: string;
  image_alt: string;
}

function ShelfItem(props: Item): React.ReactElement {
  return (
    <div className="shelf-item">
      <div className="shelf-item--header">
        <div className="shelf-item--info">
          <h1> {props.name.toUpperCase()} </h1>
          <p>{props.description}</p>
        </div>
        <div className="shelf-item--price">R$ {props.price}</div>
      </div>
      <img
        className="shelf-item--picture"
        src={props.image}
        alt={props.image_alt}
      />
    </div>
  );
}

export default ShelfItem;
