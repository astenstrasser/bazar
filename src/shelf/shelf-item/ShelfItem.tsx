import React from "react";
import "./ShelfItem.css";
import { Product } from "../../common/types";

function ShelfItem(product: Product): React.ReactElement {
  return (
    <div className="shelf-item">
      <div className="shelf-item--header">
        <div className="shelf-item--info">
          <h1> {product.name.toUpperCase()} </h1>
          <p>{product.details}</p>
        </div>
        <div className="shelf-item--price">R$ {product.price}</div>
      </div>
      <img
        className="shelf-item--picture"
        src={product.imageUrl}
        alt={product.imageAltText}
      />
    </div>
  );
}

export default ShelfItem;
