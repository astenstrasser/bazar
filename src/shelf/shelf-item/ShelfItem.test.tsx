import React from "react";
import { render } from "@testing-library/react";
import ShelfItem from "./ShelfItem";

test("renders snapshot", () => {
  const { container } = render(
    <ShelfItem
      _id="id"
      name="test"
      details="test"
      imageUrl="scr"
      price={1342.0}
      imageAltText="test"
    />
  );
  expect(container.firstChild).toMatchSnapshot();
});
