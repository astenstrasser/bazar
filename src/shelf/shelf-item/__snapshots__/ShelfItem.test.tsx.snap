// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders snapshot 1`] = `
<div
  class="shelf-item"
>
  <div
    class="shelf-item--header"
  >
    <div
      class="shelf-item--info"
    >
      <h1>
         
        TEST
         
      </h1>
      <p>
        test
      </p>
    </div>
    <div
      class="shelf-item--price"
    >
      R$ 
      1342
    </div>
  </div>
  <img
    alt="test"
    class="shelf-item--picture"
    src="scr"
  />
</div>
`;
